jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocInfo.java

```diff
@@ -200,13 +207,22 @@ public String formatClusterFilterString() throws InvalidURLException {
     public Cluster findClusterInPool(String filter) throws SQLException {
         try {
             Cluster suitableCluster = null;
+            Map<Cluster, Long> clusterLoads = new HashMap<>();
             for (Cluster response :
                     clusterControllerClient
                             .listClusters(params.projectId(), params.region(), filter)
                             .iterateAll()) {
-                // TODO: decide on how to pick a cluster among cluster pool
+                if (response.hasMetrics()
```

Comment:
YARN available memory can legitimately be 0, so I don't think you should exclude clusters with no available memory. You just need to make sure your weighted random function can handle one or all clusters having 0 available memory. 

Instead, if you want to avoid clusters that don't have that metirc set, you can do something like`!response.getMetrics().getAllYarnMetrics().containsKey(YARN_MEMORY)`